services/indexer/abstract_indexer.js

```diff
@@ -44,67 +44,88 @@ class AbstractIndexer {
     );
   }
 
-  deleteIndex(callback) {
+  deleteIndex() {
     this.logger.info(`Deleting index (${this.esIndex}).`);
-    this.client.indices.delete({
-      index: this.esIndex
-    }, (err, response, status) => {
-      if (err) {
-        this.logger.error(err);
-      }
-      this.logger.debug(status);
-      return callback(err, response);
+    return new Promise((fulfill, reject) => {
+      this.client.indices.delete({
+        index: this.esIndex
+      }, (err, response, status) => {
+        if (err) {
+          this.logger.error(err);
+          reject(err);
+        } else {
+          this.logger.debug(status);
+          fulfill(response);
+        }
+      });
     });
   }
 
-  initIndex(callback) {
+  initIndex() {
     this.logger.info(`Creating index (${this.esIndex}).`);
-    this.client.indices.create({
-      index: this.esIndex,
-      body: this.esSettings
-    }, (err, response, status) => {
-      if(err) {
-        this.logger.error(err);
-      }
-      this.logger.debug(status);
-      return callback(err, response);
+    return new Promise((fulfill, reject) => {
+      this.client.indices.create({
+        index: this.esIndex,
+        body: this.esSettings
+      }, (err, response, status) => {
+        if(err) {
+          this.logger.error(err);
+          reject(err);
+        } else {
+          this.logger.debug(status);
+          fulfill(response);
+        }
+      });
     });
   }
 
-  indexExists(callback) {
-    this.client.indices.exists({
-      index: this.esIndex
-    }, (err, response, status) => {
-      if(err) {
-        this.logger.error(err);
-      }
-      this.logger.debug(status);
-      return callback(err, response);
+  indexExists() {
+    return new Promise((fulfill, reject) => {
+      this.client.indices.exists({
+        index: this.esIndex
+      }, (err, response, status) => {
+        if(err) {
+          this.logger.error(err);
+          reject(err);
+        } else {
+          this.logger.debug(status);
+          fulfill(response);
+        }
+      });
     });
   }
 
-  indexDocument(doc, callback) {
-    this.client.index(doc, (err, response, status) => {
-      if(err) {
-        this.logger.error(err);
-      }
-      this.logger.debug(status);
-      return callback(err, response);
+  indexDocument(doc) {
+    return new Promise((fulfill, reject) => {
+      this.client.index(doc, (err, response, status) => {
+        if(err) {
+          this.logger.error(err);
+          reject(err);
+        } else {
+          this.logger.debug(status);
+          fulfill({response, status});
```

Comment:
The other seems to be translations, but this is giving back the status as well. I assume this is intentional, but just thought I'd check.